torch/csrc/jit/passes/alias_analysis.cpp

```diff
@@ -307,15 +321,41 @@ void AliasDb::analyzeImpl(Node* node) {
       if (tryRegisteredAnalysis(node)) {
         return;
       }
-      TORCH_INTERNAL_ASSERT(!aliasAnalysisHasSpecialCaseFor(node->kind()));
   }
 
-  const auto& schema = node->schema();
-  // see [custom operator aliasing]
-  if (!node->kind().is_aten() && !node->kind().is_prim()) {
+  TORCH_INTERNAL_ASSERT(op, "We should have an op schema if we get to here");
+  const AliasAnalysisKind analysis = op->aliasAnalysisKind();
+  TORCH_INTERNAL_ASSERT(analysis != AliasAnalysisKind::INTERNAL_SPECIAL_CASE && !aliasAnalysisHasSpecialCaseFor(node->kind()), "Special cases should be handled already if we're here.");
+
+  if (node->kind().is_aten() || node->kind().is_prim()) {
+    // TODO This assert is only introduced to check that we don't break the current code base. Remove this later to allow aten:: and prim:: ops to use other alias analysis kinds.
+    TORCH_INTERNAL_ASSERT(analysis == AliasAnalysisKind::FROM_SCHEMA, "aten:: and prim:: operators should use AliasAnalysisKind::FROM_SCHEMA but ", node->kind().toDisplayString(), " doesn't.");
+  }
+
+  if (analysis == AliasAnalysisKind::CONSERVATIVE) {
+    TORCH_INTERNAL_ASSERT(!node->kind().is_aten() && !node->kind().is_prim(), "aten:: and prim:: operators should not use AliasAnalysisKind::CONSERVATIVE but ", node->kind().toDisplayString(), " does.");
+
+    // TODO A previous implementation of alias analysis always accessed node->schema ,
+    // which cause the schema caches in the Node class to be filled for the full graph.
+    // Unfortunately, our JIT passes started relying on that, so we need to keep doing this.
+    // Details: in caffe2/torch/onnx/utils.py, _jit_pass_onnx is called on an invalid
+    // JIT graph because we called _jit_pass_erase_number_types right before and
+    // ints are now Tensors instead. So if _jit_pass_onnx tries to look up operator schemas,
+    // it will crash. However, _jit_pass_constant_propagation, which is called before it,
+    // runs alias analysis and prefills the schema cache in the all Node instances
+    // so that _jit_pass_onnx doesn't look up operators to get the schemas anymore.
+    // We should fix this.
+    node->schema(); // fill the schema cache in the Node class
+
     return analyzeConservative(node);
   }
 
+  TORCH_INTERNAL_ASSERT(analysis == AliasAnalysisKind::FROM_SCHEMA, "AliasAnalysisKind::CONSERVATIVE/PURE/INTERNAL_SPECIAL_CASE should already have been handled above");
+  const auto& schema = node->schema();
+
+  // TODO This assert is only introduced to check that we don't break the current code base. Remove this later to allow other ops to use AliasAnalysisKind::FROM_SCHEMA
+  TORCH_INTERNAL_ASSERT(node->kind().is_prim() || node->kind().is_aten(), "The current code base should only have AliasAnalysisKind::FROM_SCHEMA for aten:: and prim:: ops but we found it for ", node->kind().toDisplayString(), ". We want to open this up though.");
```

Comment:
Isn't this check the same as the one above?

Reply:
not sure which one above you mean, maybe the other FROM_SCHEMA one? That one asserts that if it is aten or prim, then it is FROM_SCHEMa. This one asserts that if is is FROM_SCHEMA, then it is aten or prim. Both sides of the implication are needed.